windows/table.cpp

```diff
@@ -0,0 +1,295 @@
+	LVCOLUMNW lvc;
+
+	if (c->modelColumn >=0) {
+		return; // multiple parts not implemented
+	}
+	c->modelColumn = modelColumn;
+
+	// work out appropriate listview index for the column
+	for (auto candidate : t->columns) {
+		if (candidate == c) {
+			break;
+		}
+		if (candidate->modelColumn >= 0) {
+			lvIndex++;
+		}
+	}
+
+	ZeroMemory(&lvc, sizeof (LVCOLUMNW));
+	lvc.mask = LVCF_FMT | LVCF_WIDTH | LVCF_TEXT;	/* | LVCF_SUBITEM; */
+	lvc.fmt = LVCFMT_LEFT;
+	lvc.cx = 120;	// TODO
+	lvc.pszText = c->name;
+	if (SendMessageW(c->t->hwnd, LVM_INSERTCOLUMN, (WPARAM) lvIndex, (LPARAM) (&lvc)) == (LRESULT) (-1))
+		logLastError(L"error calling LVM_INSERTCOLUMN in uiTableColumnPartSetTextPart()");
+}
+
+void uiTableColumnAppendImagePart(uiTableColumn *c, int modelColumn, int expand)
+{
+	// not implemented
+}
+
+void uiTableColumnAppendButtonPart(uiTableColumn *c, int modelColumn, int expand)
+{
+	// not implemented
+}
+
+void uiTableColumnAppendCheckboxPart(uiTableColumn *c, int modelColumn, int expand)
+{
+	// not implemented
+}
+
+void uiTableColumnAppendProgressBarPart(uiTableColumn *c, int modelColumn, int expand)
+{
+	// not implemented
+}
+
+void uiTableColumnPartSetEditable(uiTableColumn *c, int part, int editable)
+{
+	// TODO
+}
+
+void uiTableColumnPartSetTextColor(uiTableColumn *c, int part, int modelColumn)
+{
+	// not implemented
+}
+
+// uiTable implementation
+
+uiWindowsControlAllDefaultsExceptDestroy(uiTable)
+
+uiTableColumn *uiTableAppendColumn(uiTable *t, const char *name)
+{
+	uiTableColumn *c;
+   
+	c = uiprivNew(uiTableColumn);
+	c->name = toUTF16(name);
+	c->t = t;
+	c->modelColumn = -1;	// -1 = unassigned
+	// we defer the actual ListView_InsertColumn call until a part is added...
+	t->columns.push_back(c);
+	return c;
+}
+
+void uiTableSetRowBackgroundColorModelColumn(uiTable *t, int modelColumn)
+{
+	// not implemented
+}
+
+static void uiTableDestroy(uiControl *c)
+{
+	uiTable *t = uiTable(c);
+	uiTableModel *model = t->model;
+	std::vector<uiTable *>::iterator it;
+
+	uiWindowsUnregisterWM_NOTIFYHandler(t->hwnd);
+	uiWindowsEnsureDestroyWindow(t->hwnd);
+	// detach table from model
+	for (it = model->tables.begin(); it != model->tables.end(); it++) {
+		if (*it == t) {
+			model->tables.erase(it);
+			break;
+		}
+	}
+	// free the columns
+	for (auto col : t->columns) {
+		uiprivFree(col->name);
+		uiprivFree(col);
+	}
+	t->columns.~vector<uiTableColumn *>();	// (created with placement new, so just call dtor directly)
+	uiFreeControl(uiControl(t));
+}
+
+// suggested listview sizing from http://msdn.microsoft.com/en-us/library/windows/desktop/dn742486.aspx#sizingandspacing:
+// "columns widths that avoid truncated data x an integral number of items"
+// Don't think that'll cut it when some cells have overlong data (eg
+// stupidly long URLs). So for now, just hardcode a minimum:
+#define tableMinWidth 107		/* in line with other controls */
+#define tableMinHeight (14*3)	/* header + 2 lines (roughly) */

```

Comment:
There might be a way to use `LVM_GETHEADER` to get the necessary size of the header control, perhaps combined with `HDM_LAYOUT`, but I'm not sure how this cooperates with listview controls. Mark this as a TODO for now.